ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetFilteredObjects.java

```diff
@@ -0,0 +1,131 @@
+        this.filterParams = args.getFilterParams();
+        this.bucketName = args.getBucket();
+
+        if (!Guard.isStringNullOrEmpty(args.getPrefix())) {
+            throw new BadArgumentException("'-p' prefix is not supported.");
+        }
+        this.prefix = null;
+        return this;
+    }
+
+    @Override
+    public GetDetailedObjectsResult call() throws Exception {
+
+        final ImmutableList<DetailedS3Object> suspectBulkObjects;
+        Predicate<DetailedS3Object> filterPredicate;
+
+        if (Guard.isMapNullOrEmpty(this.filterParams)) {
+            // no filter params specified, run wide open
+            filterPredicate = new Predicate<DetailedS3Object>() {
+                @Override
+                public boolean apply(@Nullable final DetailedS3Object input) {
+                    return (input != null);
+                }};
+        } else {
+            long larger = 0L;
+            long smaller = Long.MAX_VALUE;
+            Date newer = new Date(0);
+            Date older = new Date(Long.MAX_VALUE);
+
+            for (final String paramChange : this.filterParams.keySet()) {
+                final String paramNewValue = this.filterParams.get(paramChange);
+                if ("largerthan".equalsIgnoreCase(paramChange)) {
+                    larger = Long.parseLong(paramNewValue);
+                } else if ("smallerthan".equalsIgnoreCase(paramChange)) {
+                    smaller = Long.parseLong(paramNewValue);
+                } else if ("newerthan".equalsIgnoreCase(paramChange)) {
+                    newer = new Date(new Date().getTime() - Utils.dateDiffToSeconds(paramNewValue) * 1000);
+                } else if ("olderthan".equalsIgnoreCase(paramChange)) {
+                    older = new Date(new Date().getTime() - Utils.dateDiffToSeconds(paramNewValue) * 1000);
+                } else {
+                    throw new CommandException("Unrecognized filter parameter: " + paramChange);
+                }
+            } // for
+
+            // make final for inner class
+            final long largerthan = larger;
+            final long smallerthan = smaller;
+            final Date newerthan = newer;
+            final Date olderthan = older;
+
+            filterPredicate = new Predicate<DetailedS3Object>() {
+                @Override
+                public boolean apply(@Nullable final DetailedS3Object input) {
+                    if (input == null) {
+                        return false;
+                    }
+                    return (input.getSize() > largerthan
+                            && input.getSize() < smallerthan
+                            && input.getCreationDate().after(newerthan)
+                            && input.getCreationDate().before(olderthan)
+                    );
+                }
+            };
+        }
+
+        // get filtered list using pagination
+        suspectBulkObjects = ImmutableList.copyOf(
```

Comment:
Since you end up calling `.iterator()` on `suspectBulkObjects` when you pass it to `GetDetailedObjectsResult` you don't need to convert the `FluentIterable` into a list.